examples/examples/z_formats.rs

```diff
@@ -0,0 +1,16 @@
+use zenoh::{keformat, prelude::keyexpr};
+
+keformat!("user_id/${user_id:*}/file/${file:**}", format);
+
+fn main() {
+    let mut formatter = format::formatter();
+    let file = "hi/there";
+    let ke = keformat!(formatter, user_id = 42, file)
+        .unwrap()
```

Comment:
Maybe I'm dumb, but why this can fail twice?
I imagine that the first one is when the `formatter` used does not match the parameters (or vice-versa).
But why the `build()` can fail?

Reply:
This can fail if some of the fields haven't received values yet, but expected non-empty values